src/getEdgeMassMatrix.jl

```diff
@@ -1,54 +1,291 @@
-export getEdgeMassMatrix, getdEdgeMassMatrix, getEdgeMassMatrixNoWeight
+export getEdgeMassMatrix, getdEdgeMassMatrix
 
 
+"""
 function getEdgeMassMatrix(M::OcTreeMeshFV,sigma::Vector)
-    if isempty(M.Pe)
-        M.Pe = getEdgeMassMatrixAnisotropic(M.S,M.h)
-    end
-	M = M.Pe' * kron(speye(24),sdiag(sigma)) * M.Pe
-	return M
+
+Returns finite volume edge mass matrix. Can handle isotropic and anisotropic models.
+
+
+Input:
+   M::OcTreeMeshFV Octree mesh object
+   sigma::vector Conductivity model. Can be used for isotropic and anisotropic models depending on length
+          of sigma. For an isotropic model length(sigma) = nc, where nc is the number of mesh cells.
+          Use length(sigma)=3*nc a diagonally anisotropic model and length(sigma)=6*nc for 
+          for fully anisotropic model.
+Output:
+   Me::SparseMatrixCSC Edge mass matrix
+
+"""
+function getEdgeMassMatrix(M::OcTreeMeshFV,sigma::Vector)
+  # For octree meshes
+  if isempty(M.Pe)
+    M.Pe = getEdgeMassMatrixIntegrationMatrix(M.S, M.h)
+  end
+  P = M.Pe
+  n = length(sigma)
+  if n == M.nc
+    M = P' * kron(speye(24),spdiagm(sigma)) * P
+  elseif n == 3 * M.nc
+    M = P' * kron(speye(8),spdiagm(sigma)) * P
+  elseif n == 6 * M.nc
+    R12 = kron(sparse([2,1,3],[1,2,3],1),speye(Int,M.nc))
+    R23 = kron(sparse([1,3,2],[1,2,3],1),speye(Int,M.nc))
+    D   = spdiagm(sigma[       1:3*M.nc])
+    N   = spdiagm(sigma[3*M.nc+1:6*M.nc])
+    S   = D + R12 * N * R23 + R23 * N * R12
+    M   = P' * kron(speye(8),S) * P
+   else
+     error("Invalid size of sigma")
+   end
+	 return M
 end
 
-function getEdgeMassMatrixNoWeight(M::OcTreeMeshFV,sigma::Vector)
-    P = getEdgeMassMatrixAnisotropicNoWeight(M.S,M.h)    
-	M = P'* kron(speye(24),sdiag(sigma)) *P
-	return M
+function getdEdgeMassMatrix(M::OcTreeMeshFV,sigma::Vector,v::Vector)
+  # Derivative
+  if isempty(M.Pe)
+    M.Pe = getEdgeMassMatrixIntegrationMatrix(M.S, M.h)
+  end
+  P = M.Pe
+  w = P * v
+  n = length(sigma)
+  if n == M.nc
+    dM = P' * spdiagm(w) * kron(ones(24),speye(M.nc))
+  elseif n == 3 * M.nc
+    dM = P' * spdiagm(w) * kron(ones(8),speye(3*M.nc))
```

Comment:
Multiplication of the transpose of a sparse matrix with a diagonal matrix (what we would have with `P' * Diagonal(w)`) doesn't work due to a bug in Julia. See the issue (https://github.com/JuliaLang/julia/issues/17456) for an explanation. There is an open PR with a fix but it hasn't been merged: https://github.com/JuliaLang/julia/pull/17507.